packages/device_info/device_info_platform_interface/lib/model/android_device_info.dart

```diff
@@ -113,34 +113,40 @@ class AndroidDeviceInfo {
   /// Deserializes from the message received from [_kChannel].
   static AndroidDeviceInfo fromMap(Map<String, dynamic> map) {
     return AndroidDeviceInfo(
-      version:
-          AndroidBuildVersion._fromMap(map['version']!.cast<String, dynamic>()),
-      board: map['board']!,
-      bootloader: map['bootloader']!,
-      brand: map['brand']!,
-      device: map['device']!,
-      display: map['display']!,
-      fingerprint: map['fingerprint']!,
-      hardware: map['hardware']!,
-      host: map['host']!,
-      id: map['id']!,
-      manufacturer: map['manufacturer']!,
-      model: map['model']!,
-      product: map['product']!,
-      supported32BitAbis: _fromList(map['supported32BitAbis']!),
-      supported64BitAbis: _fromList(map['supported64BitAbis']!),
-      supportedAbis: _fromList(map['supportedAbis']!),
-      tags: map['tags']!,
-      type: map['type']!,
-      isPhysicalDevice: map['isPhysicalDevice']!,
-      androidId: map['androidId']!,
-      systemFeatures: _fromList(map['systemFeatures']!),
+      version: AndroidBuildVersion._fromMap(map['version'] != null
+          ? map['version'].cast<String, dynamic>()
+          : <String, dynamic>{}),
+      board: map['board'] ?? '',
+      bootloader: map['bootloader'] ?? '',
+      brand: map['brand'] ?? '',
+      device: map['device'] ?? '',
```

Comment:
The empty string means that there's a bug on the platform side right? I think we should have a tracking bug to address this with Pigeon, and add TODOs.   We should really try to fix this with Pigeon because I can see this being a source of bugs.

Reply:
Why does it have anything to do with Pigeon? This plugin doesn't use pigeon.
Anyways, from dart perspective, we should always assume the method channel could return null. Even the native API doesn't return null now, it could in the future as we don't use null safe language on native side.

Reply:
I guess I was referring to the Dart API. If the empty string is documented on the Android official website, then it's correct to provide empty string values.  The point about Pigeon is about making this enforcement more explicit upstream, so you don't have to manually deal with it in Dart.

Reply:
>> The point about Pigeon is about making this enforcement more explicit upstream, so you don't have to manually deal with it in Dart.

Did you mean when we eventually migrate this plugin to pigeon?

>> If the empty string is documented on the Android official website, then it's correct to provide empty string values.

It's not documented. I also don't want to make everything nullable either. I think in a null safe language, it is usually the case that we don't want to make a string nullable if empty string doesn't have a specific meaning.

Reply:
yes, eventual migration. The suggestion is to look at the official documentation for explicit mentions of "null". For example, the bug that Jason was fixing in Java. https://developer.android.com/reference/android/content/pm/FeatureInfo#name The docs explicitly say that it can be `null`. 

I think that is the type of work that will match the intended behavior of the platform.

Reply:
Most of the properties in the documentation doesn't say it is nullable or non-null. See https://developer.android.com/reference/kotlin/android/os/Build 

So since they are strings, I'd prefer making them non-null and empty string for "no value" on dart side.